packages/edit-navigation/src/components/name-editor/index.js

```diff
@@ -2,34 +2,58 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { useEffect, useState } from '@wordpress/element';
+import { useEffect, useRef, useState } from '@wordpress/element';
 /**
  * Internal dependencies
  */
-import { BlockControls } from '@wordpress/block-editor';
-import { ToolbarGroup, ToolbarItem } from '@wordpress/components';
+import { InspectorControls } from '@wordpress/block-editor';
+import { PanelBody, BaseControl } from '@wordpress/components';
 import { useMenuEntity, useNavigationEditorMenu } from '../../hooks';
+import { useInstanceId } from '@wordpress/compose';
 
-export function NameEditor() {
+export function NameEditor( {
+	setIsMenuNameEditFocused,
+	isMenuNameEditFocused,
+} ) {
 	const { menuName, menuId } = useNavigationEditorMenu();
 	const { editMenuName } = useMenuEntity( menuId );
+	const inputRef = useRef();
 	const [ tmpMenuName, setTmpMenuName ] = useState( menuName );
+	const instanceId = useInstanceId( NameEditor );
+	const id = `components-edit-navigation-name-editor__input-${ instanceId }`;
 	useEffect( () => setTmpMenuName( menuName ), [ menuName ] );
+	useEffect( () => {
+		if ( isMenuNameEditFocused ) inputRef.current.focus();
+	}, [ isMenuNameEditFocused ] );
 	return (
 		<>
-			<BlockControls>
-				<ToolbarGroup>
-					<ToolbarItem
-						as="input"
-						value={ tmpMenuName }
-						onChange={ ( { target: { value } } ) => {
-							setTmpMenuName( value );
-							editMenuName( value );
-						} }
-						aria-label={ __( 'Edit menu name' ) }
-					/>
-				</ToolbarGroup>
-			</BlockControls>
+			<InspectorControls>
+				<PanelBody title={ __( 'Menu Settings' ) }>
+					<BaseControl label={ __( 'Name' ) } id={ id }>
```

Comment:
currently, there is a known bug: this item is changing its place in a navbar, once a save button is hit. I am searching for a solution for it.

Reply:
If you save the name goes after the delete button, but if you then select the navigation block via the parent select button, the name appears correctly. I guess the way saving triggers rerendering plus the fact that some inspector itemsa are added by `InspectorAdditions` while others via this `NameEditor` filter is the cause.

When the name is changed, only the block is re-rendered, not the entire editor. So `InspectorAdditions` is not re-rendered. Maybe we could add `InspectorAdditions` via the same filter?

Reply:
Yeah, I think using the same filter for the menu name and delete button would probably solve it, if that's possible.